db/commitlog/commitlog.cc

```diff
@@ -1535,6 +1535,7 @@ future<db::commitlog::segment_manager::sseg_ptr> db::commitlog::segment_manager:
         co_await f.close();
     }
     if (ep) {
+        add_file_to_delete(filename, d);
```

Comment:
Will add_file_to_delete tolerate a non-existing file? In case we failed immediately on creation.

Are we guaranteed never to generate the same file name again? So we don't later delete a legitimate file.

Reply:
We are guaranteed to not re-generate the file name, yes. Non-existent should be fine, it will potentially generate an error message + extra notification in delete_segments, but that should be harmless. But it should resolve itself, and easier than trying to keep exact tabs here, since the problem could also be external - i.e. someone deleted a file on the recycled list for example.

Reply:
Oh I agree. Just want to make sure we don't trip on the exception and isolate the node.

Reply:
It is probably worth adding a test for. I'll add one and push a v2.

Reply:
Too late, I queued, so please add the test as a separate patch.